javalib/src/main/scala/java/time/Duration.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2007-present, Stephen Colebourne & Michael Nascimento Santos
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither the name of JSR-310 nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package java.time
+
+import java.time.temporal.TemporalAmount
+
+@SerialVersionUID(3078945930695997490L)
+final class Duration private (private val seconds: Long, private val nanos: Int)
+    extends TemporalAmount
+    with Ordered[Duration]
+    with Serializable {
+
+  def toMillis: Long = {
+    val result: Long = Math.multiplyExact(seconds, Duration.MILLIS_PER_SEC)
```

Comment:
I believe current ccala best practice is to drop the  `: Long` type declaration?/ascription?   
It is probably not wrong, but is is also not Haskell minimalism.

I often use the type declaration in my development phase or debugging phase where I want
to ensure that the Right Hand Side is generating the type I expect.  I have been trained to delete
that code for final review.

Others may know better

Reply:
On public definitions a return type is desired. I think too since procedure syntax going away that it is desired to add a return type to help the type safety and readability.

Reply:
Sorry for my poor quoting skills. I was trying to refer to the line:
` val result: Long = Math.multiplyExact(seconds, Duration.MILLIS_PER_SEC)`.

This general idea happens in several places in the file.  Probably not worth another
edit & CI cycle.  I mentioned it because I was asked to be in super-picky line edit mode.

I concur that the return type in the def is indeed needed.

Reply:
Oh, I see. I am not really sure if that is ok style or not but it is defensive for sure. If @densh doesn't like it I could still make some changes.